### PR TITLE
Add support for Groups API method transfer.

### PR DESCRIPTION
**Pull request by sattlerc**

### CODE REVIEW

**Comment by JohnVillalovos** [comment]
Thanks for the PR. Can you please squash the commits down to one commit. And make sure there is a commit message.

Also if possible unit and/or functional tests would be great.

Thanks again!

**Comment by JohnVillalovos** [comment]
And updating the documentation on how to use it would also be useful.

**Comment by sattlerc** [comment]
Note that the typo fix commit is unrelated to the feature commit. Should I remove it?

I looked at the documentation, but it seems to be autogenerated.

There are usage examples for a limited number of group methods in `docs/gl_objects/groups.rst`. The related group method `transfer_project` doesn't appear there, so I thought it appropriate not to add `transfer_group`.

File: gitlab/v4/objects/groups.py
Comment (JohnVillalovos) [inline gitlab/v4/objects/groups.py:118]: Does this work? I'm not sure why it is a function.

Also when `group_id` is `None` it returns `None`, but `dict(None)` doesn't work so it would fail on L118.

```suggestion
        post_data = {}
        if group_id is not None:
            post_data['group_id"] = group_id

        self.manager.gitlab.http_post(path, post_data=post_data, **kwargs)
```

Tests to validate this works would be great.

**Review by JohnVillalovos — CHANGES_REQUESTED** [review]

File: gitlab/v4/objects/groups.py
Comment (sattlerc) [reply gitlab/v4/objects/groups.py:118]: It's functional style. When `group_id` is `None`, it returns the empty iterator, not `None`, so works correctly.

I'll use the imperative style.

Comment (JohnVillalovos) [reply gitlab/v4/objects/groups.py:118]: Thanks! Make it easier for our little brains on this side 😊

**Comment by sattlerc** [comment]
I wasn't able to find tests for the related group method `transfer_project`. If you add tests for that one, I can base the ones for `transfer_group` on those.

**Review by JohnVillalovos — APPROVED** [review]

File: gitlab/v4/objects/groups.py
Comment (nejch) [inline gitlab/v4/objects/groups.py:99]:
```suggestion
        """Transfer the group to a new parent group or turn the subgroup into
        a top-level group.
```

**Review by nejch — COMMENTED** [review]
Thanks for this PR @sattlerc! Just another small comment from me.

thought: I also would almost prefer to just call it `transfer()` since the transfer direction is inverted from the `transfer_project()` in this same class, but unfortunately we have `transfer_project()` in the projects module already as well. Maybe that one could be deprecated and have both methods that do "transfer _this_ resource somewhere" called `transfer()`. Outside the scope of this PR but open to hear opinions here :)

And thanks for changing the functional code as this does look closer to the rest of the library e.g. further down in this module.
https://github.com/python-gitlab/python-gitlab/blob/9894b3580a7eb5c2e377c482820ff3210f913abe/gitlab/v4/objects/groups.py#L326-L331

**Comment by nejch** [comment]
> I wasn't able to find tests for the related group method `transfer_project`. If you add tests for that one, I can base the ones for `transfer_group` on those.

I'll try and add some tests for `transfer_project` as well.

**Comment by sattlerc** [comment]
I would prefer `transfer` as well. Should I change it?

File: gitlab/v4/objects/groups.py
Comment (sattlerc) [reply gitlab/v4/objects/groups.py]: Fixed.

**Comment by nejch** [comment]
> I would prefer `transfer` as well (that's also the name of the API path). Should I change it?

Yeah, let's do that. This just looks nicer:

```python
group = gl.groups.get("gitlab-org/maintainers")
group.transfer("gitlab-com")
```

It also avoids us getting into an awkward situation if GitLab comes up with a "transfer a group into this group" endpoint at some point :sweat_smile: (as is the case with `transfer_project` here)

**Comment by sattlerc** [comment]
Okay. I have have also renamed the exception from `GitlabTransferGroupError` to `GitlabGroupTransferError`.

**Comment by nejch** [comment]
Thanks, LGTM @sattlerc, not sure if John has any more comments. I'll open an accompanying PR for project transfer and you can rebase after merge if you like, I'll ping here.

**Comment by JohnVillalovos** [comment]
> not sure if John has any more comments. 

Only that tests would be nice. But I think you said you will work on setting up the test framework and then we can land this on top of that.

**Comment by sattlerc** [comment]
Don't worry about it.

**Pull request closed without merging**
